### DISCUSSION TITLE
alt text for images that come from code cells?

### DISCUSSION THREAD

**Discussion by ericvd-ucb** [executablebooks/meta#1155] [General]
In our book, there are many figures that come as a result of running a code cell, Eg a data visualization. - but we cant fiture out how to make alt text for that   ( the alt text for an inserted figure is the kind of thing we are hoping for)

**Comment by rowanc1** [comment]
Are you using JupyterBook or mystmd.org to render your book?

Perhaps related: https://github.com/executablebooks/mystmd/pull/1077

**Reply by ericvd-ucb** [reply]
we are using Jupyterbook  - https://data88e.org/textbook/

Do I need to move to MystMD to get better alt text tags?

Interesting that @choldgraf asked a similar question at the same
Seems like the feature is under development

**Comment by choldgraf** [comment]
If you're using the Sphinx stack, this should be doable with the `{glue:figure}` directive for notebook outputs:

https://myst-nb.readthedocs.io/en/latest/render/glue.html#the-glue-figure-directive

The `figcaption` should be a more a11y-friendly way of displaying the image. Would that work?

In the long-term, we're shifting more and more effort into [the new MyST document engine](https://mystmd.org) and at some point we want to prototype allowing Jupyter Book to use that engine under the hood instead of Sphinx.

**Comment by pancakereport** [comment]
Looks like we found something that seems to work (though it's annoying as you save the figure and then pop it in with a directive, similar to what Chris was describing.. I think). Answering because this question does pop up fairly high in my Google results.

Source: https://github.com/data-88e/textbook/blob/master/content/01-demand/01-demand.ipynb
Deployed book: https://data88e.org/textbook/content/demand/demand/
